Documentation/installing-dependencies.rst

```diff
@@ -1,216 +1,293 @@
   cd ..
 
-Installing netCDF
-~~~~~~~~~~~~~~~~~
+Installing Eigen
+""""""""""""""""
 
-.. code-block:: bash
-
-  wget https://syncandshare.lrz.de/dl/fiJNAokgbe2vNU66Ru17DAjT/netcdf-4.6.1.tar.gz
-  tar -xaf netcdf-4.6.1.tar.gz
-  cd netcdf-4.6.1
-  CFLAGS="-fPIC ${CFLAGS}" CC=h5pcc ./configure --enable-shared=no --prefix=$HOME --disable-dap
-  #NOTE: Check for this line to make sure netCDF is built with parallel I/O: 
-  #"checking whether parallel I/O features are to be included... yes" This line comes at the very end (last 50 lines of configure run)!
-  make -j8
-  make install
-  cd ..
-
-.. _installing_eigen3:
-
-Installing Eigen3
-~~~~~~~~~~~~~~~~~
+Next, we look at Eigen which conveniently uses CMake as a build system for itself.
+Eigen is used in SeisSol for setting up matrices and other numerical computations, and optionally, also as code generator for matrix chain products.
+Once again, if you do not have Eigen installed, you may do so manually as follows:
 
 .. code-block:: bash
 
    wget https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz
    tar -xf eigen-3.4.0.tar.gz
    cd eigen-3.4.0
    mkdir build && cd build
-   cmake .. -DCMAKE_INSTALL_PREFIX=~
+   cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX
    make install
    cd ../..
 
-.. _installing_libxsmm:
+Lastly, we need easi which is (most likely) not already installed on your system or as a module file, as it is a more SeisSol-specific library.
+It is used for setting up the model parameters.
+Here you can find the `installation instructions <https://easyinit.readthedocs.io/en/latest/getting_started.html>`_.
+
+And with that, we're good to go!
+
+Additional Requirements for GPUs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For GPUs, we need some more packages.
+
+Installing SYCL (for GPUs)
+""""""""""""""""""""""""""
+
+See section :ref:`Installing SYCL <installing_SYCL>`.
+
+Installing GemmForge, ChainForge (for GPUs)
+"""""""""""""""""""""""""""""""""""""""""""
+
+.. _gemmforge_installation:
+
+The GPU code generators are called GemmForge and ChainForge.
+Conveniently, they come as Python packages and can be installed with the following commands.
+
+.. code-block:: bash
+
+   pip3 install --user git+https://github.com/SeisSol/gemmforge.git
+   pip3 install --user git+https://github.com/SeisSol/chainforge.git
+
+Note that ChainForge is optional.
+
+Once you have SYCL and GemmForge (maybe also ChainForge) ready, you are set for compiling SeisSol with GPUs.
+
+Code Generators for CPUs (optional, recommended)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For CPU code generators, we support the following:
+
+- Eigen
+- libxsmm (libxsmm\_gemm\_generator) for small matrix multiplications
+- PSpaMM (pspamm.py) for small sparse matrix multiplications (required only on Knights Landing or Skylake)
+
+Note that using Eigen does not result in any additional dependencies, since it is needed in SeisSol anyways.
+Other than that, we recommend using the combination libxsmm and PSpaMM.
```

Comment:
We recommend `libxsmm` and in addition `PSpaMM` for architectures with `AVX-512`.